code/source/playfab/PlayFabHCHttpPlugin.cpp

```diff
@@ -0,0 +1,166 @@
+#include <stdafx.h>
+#include <playfab/PlayFabHCHttpPlugin.h>
+#include <playfab/PlayFabSettings.h>
+
+namespace PlayFab
+{
+    // RAII wrapper around HCCallHandle
+    class HCHttpCall
+    {
+    public:
+        static void Perform(std::unique_ptr<CallRequestContainer> requestContainer);
+
+        virtual ~HCHttpCall() noexcept;
+
+    private:
+        HCHttpCall() = default;
+        HCHttpCall(const HCHttpCall& other) = delete;
+        HCHttpCall& operator=(HCHttpCall other) = delete;
+
+        HRESULT Initialize(std::unique_ptr<CallRequestContainer> requestContainer);
+        void CompleteRequest();
+
+        static void CALLBACK HCPerformComplete(XAsyncBlock* async);
+
+        HCCallHandle m_callHandle{ nullptr };
+        std::unique_ptr<CallRequestContainer> m_requestContainer;
+        XAsyncBlock m_asyncBlock{};
+    };
+
+    void PlayFabHCHttpPlugin::MakePostRequest(std::unique_ptr<CallRequestContainerBase> requestContainer)
+    {
+        CallRequestContainer* container = dynamic_cast<CallRequestContainer*>(requestContainer.get());
+        if (container != nullptr)
+        {
+            container->ThrowIfSettingsInvalid();
+            requestContainer.release();
+            HCHttpCall::Perform(std::unique_ptr<CallRequestContainer>{ container });
+        }
+    }
+
+    size_t PlayFabHCHttpPlugin::Update()
+    {
+        throw PlayFabException(PlayFabExceptionCode::ThreadMisuse, "You should not call Update() when using XAsync");
+    }
+
+    HCHttpCall::~HCHttpCall() noexcept
+    {
+        if (m_callHandle)
+        {
+            HCHttpCallCloseHandle(m_callHandle);
+        }
+    }
+
+    void HCHttpCall::Perform(std::unique_ptr<CallRequestContainer> requestContainer)
+    {
+        std::unique_ptr<HCHttpCall> call{ new HCHttpCall };
+        HRESULT hr = call->Initialize(std::move(requestContainer));

```

Comment:
mem hook here and elsewhere in file

Reply:
I did do it in some spots but its far from done.  Will be sure to do it eventually.

---
In reply to: [590802689](https://github.com/jplafonta/XPlatCppSdk/pull/2#discussion_r590802689) [](ancestors = 590802689)